File: 백준/Silver/7568. 덩치/덩치.kt
fun main() {
    val N = readLine()!!.toInt()

    val result = mutableListOf<List<Int>>()

    val r = mutableListOf<Int>()

    for (i in 0 until N) {
        val n = readLine()!!.split(" ").map { it.toInt() }
        result.add(n)
    }

    for ((index, value) in result.withIndex()) {
        var cnt = 0

        for ((i, v) in result.withIndex()) {
            if (i == index) { // 본인 제외
                continue
            }
            if (value[0] < v[0] && value[1] < v[1]) { // 카운트
                cnt++
            }
        }
        r.add(cnt + 1)
    }

    for (i in r) {
        print("$i ")
    }
}

File: 백준/Silver/7568. 덩치/덩치.py
N = int(input())

result = []

r = []

for i in range(0,N):
    n  = list(map(int,input().split()))
    result.append(n)

    
for index, value in enumerate(result):

    cnt = 0
    for i, v in enumerate(result):
        if i==index:
            continue
        if value[0] < v[0] and value[1] < v[1]:
            cnt+=1
    r.append(cnt+1)

for i in r:
    print(i,end=" ")